dali/pipeline/operator/builtin/make_contiguous.cc

```diff
@@ -12,24 +12,62 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "dali/core/nvtx.h"
 #include "dali/pipeline/operator/builtin/make_contiguous.h"
 
 namespace dali {
 
 void MakeContiguousCPU::RunImpl(HostWorkspace &ws) {
   auto &input = ws.template Input<CPUBackend>(0);
   auto &output = ws.template Output<CPUBackend>(0);
-  int batch_size = input.num_samples();
-  output.SetLayout(input.GetLayout());
-  auto shapes = input.shape();
-
-  auto &thread_pool = ws.GetThreadPool();
-  for (int sample_id = 0; sample_id < batch_size; ++sample_id) {
-    thread_pool.AddWork([sample_id, &input, &output] (int tid) {
-      output.UnsafeCopySample(sample_id, input, sample_id, AccessOrder::host());
-    }, shapes.tensor_size(sample_id));
+
+  DomainTimeRange tr("[DALI][MakeContiguousCPU] H2H", DomainTimeRange::kBlue);
+  if (IsPassThrough()) {
+    output.ShareData(input);
+  } else {
+    int batch_size = input.num_samples();
+    output.SetLayout(input.GetLayout());
+    auto shapes = input.shape();
+
+    auto &thread_pool = ws.GetThreadPool();
+    for (int sample_id = 0; sample_id < batch_size; ++sample_id) {
+      thread_pool.AddWork([sample_id, &input, &output] (int tid) {
+        output.UnsafeCopySample(sample_id, input, sample_id, AccessOrder::host());
+      }, shapes.tensor_size(sample_id));
+    }
+    thread_pool.RunAll();
   }
-  thread_pool.RunAll();
+}
+
+void MarkPassThrough(OperatorBase &op) {
+  auto *make_contiguous_cpu = dynamic_cast<MakeContiguousBase<CPUBackend> *>(&op);
+  if (make_contiguous_cpu) {
+    make_contiguous_cpu->MarkPassThrough();
+  }
+  auto *make_contiguous_mixed = dynamic_cast<MakeContiguousBase<MixedBackend> *>(&op);
+  if (make_contiguous_mixed) {
+    make_contiguous_mixed->MarkPassThrough();
+  }
+  auto *make_contiguous_gpu = dynamic_cast<MakeContiguousBase<GPUBackend> *>(&op);
+  if (make_contiguous_gpu) {
+    make_contiguous_gpu->MarkPassThrough();
+  }
```

Comment:
Do you want to fail here as well? (like in `IsPassThrough`)
Or, if calling those functions on operators that are not MakeContiguous would rather be a programmer's mistake, then maybe there should be assert instead of DALI_FAIL.

Reply:
I guess I can for completeness. I was thinking about it, this way misusing it will be easily spotted.